modules/solace/solace.go

```diff
@@ -0,0 +1,291 @@
+package solace
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"os"
+	"time"
+
+	"github.com/docker/docker/api/types/container"
+	"github.com/docker/go-connections/nat"
+
+	"github.com/testcontainers/testcontainers-go"
+	"github.com/testcontainers/testcontainers-go/wait"
+)
+
+type SolaceContainer struct {
+	Container      testcontainers.Container
+	Username       string
+	Password       string
+	Vpn            string
+	queues         map[string][]string // queueName -> topics
+	WithClientCert bool
+	Image          string
+	ctx            context.Context
+	exposedPorts   []string
+	envVars        map[string]string
+	shmSize        int64
```

Comment:
suggestion: I'd say all the properties must be private, adding accessor container methods if needed: e.g. `username` would have a `Username() string` accessor.

Else, client could could change the internal state of the container at runtime